sycl/source/device.cpp

```diff
@@ -208,5 +208,78 @@ pi_native_handle device::getNative() const { return impl->getNative(); }
 
 bool device::has(aspect Aspect) const { return impl->has(Aspect); }
 
+void device::ext_oneapi_enable_peer_access(const device &peer) {
+  const RT::PiDevice Device = impl->getHandleRef();
+  const RT::PiDevice Peer = peer.impl->getHandleRef();
+  if (Device != Peer) {
+    auto Plugin = impl->getPlugin();
+    pi_result result =
+        Plugin->call_nocheck<detail::PiApiKind::piextEnablePeerAccess>(Device,
+                                                                       Peer);
+    if (result != PI_SUCCESS) {
+      char *message = nullptr;
+      Plugin->call<detail::PiApiKind::piPluginGetLastError>(&message);
+      throw sycl::exception(make_error_code(errc::runtime), message);
+    }
+  }
+}
+
+void device::ext_oneapi_disable_peer_access(const device &peer) {
+  const RT::PiDevice Device = impl->getHandleRef();
+  const RT::PiDevice Peer = peer.impl->getHandleRef();
+  if (Device != Peer) {
+    auto Plugin = impl->getPlugin();
+    pi_result result =
+        Plugin->call_nocheck<detail::PiApiKind::piextDisablePeerAccess>(Device,
+                                                                        Peer);
+    if (result != PI_SUCCESS) {
+      char *message = nullptr;
+      Plugin->call<detail::PiApiKind::piPluginGetLastError>(&message);
+      throw sycl::exception(make_error_code(errc::runtime), message);
+    }
+  }
+}
+
+bool device::ext_oneapi_can_access_peer(const device &peer,
+                                        ext::oneapi::peer_access attr) {
+  const RT::PiDevice Device = impl->getHandleRef();
+  const RT::PiDevice Peer = peer.impl->getHandleRef();
+
+  if (Device == Peer) {
+    return true;
+  }
+
+  if (peer.get_backend() != backend::ext_oneapi_cuda) {
+    return false;
+  }
+
+  RT::PiPeerAttr PiAttr;
+  size_t return_size;
+  int value;
+
+  switch (attr) {
+  case ext::oneapi::peer_access::access_supported: {
+    PiAttr = PI_PEER_ACCESS_SUPPORTED;
+    break;
+  }
+  case ext::oneapi::peer_access::atomics_supported: {
+    PiAttr = PI_PEER_ATOMICS_SUPPORTED;
+    break;
+  }
+  }
+  auto Plugin = impl->getPlugin();
+  pi_result result =
+      Plugin->call_nocheck<detail::PiApiKind::piextPeerAccessGetInfo>(
+          Device, Peer, PiAttr, sizeof(int), &value, &return_size);
+
+  if (result != PI_SUCCESS) {
+    char *message = nullptr;
+    Plugin->call<detail::PiApiKind::piPluginGetLastError>(&message);
+    throw sycl::exception(make_error_code(errc::runtime), message);
+  }
+
+  return value == 1;
+}
```

Comment:
Nit; The variable naming style in these changes are inconsistent. Could you please make sure they follow the surrounding file's style as much as possible? That also goes for other files, such as device.hpp.

Reply:
Naming convention in this file appears to be camelCase except for `RT::` stuff. So I've only updated `returnSize`, but I'll have another read through all changes.

Reply:
Looks like you're right. The file is a little inconsistent about use of camelCase and PascalCase (the latter being what the style-guide suggests.) As long as we can argue it follows the surrounding code I'm happy and we can make a pass over it to bring it in line with the style-guide later.

Reply:
pi_cuda is generally a mixture. UR looks the same. I've raised this with UR team. For the changes I made in pi_cuda I've used a consistent naming convention within the new functions. For the other PI's I've been consistent with the naming conventions used for the same var names in other getInfo functions etc.